en/examples/fly_mission_qgc_plan.md

```diff
@@ -0,0 +1,337 @@
+        handle_mission_err_exit(result, "Mission upload failed: ");
+        std::cout << "Mission uploaded." << std::endl;
+    }
+
+    std::cout << "Arming..." << std::endl;
+    const Action::Result arm_result = action->arm();
+    handle_action_err_exit(arm_result, "Arm failed: ");
+    std::cout << "Armed." << std::endl;
+
+    // Before starting the mission subscribe to the mission progress.
+    mission->subscribe_progress(
+    [](int current, int total) {
+        std::cout << "Mission status update: " << current << " / " << total << std::endl;
+    });
+
+    {
+        std::cout << "Starting mission." << std::endl;
+        auto prom = std::make_shared<std::promise<Mission::Result>>();
+        auto future_result = prom->get_future();
+        mission->start_mission_async(
+        [prom](Mission::Result result) {
+            prom->set_value(result);
+            std::cout << "Started mission." << std::endl;
+        });
+
+        const Mission::Result result = future_result.get();
+        handle_mission_err_exit(result, "Mission start failed: ");
+    }
+
+    while (!mission->mission_finished()) {
+        sleep_for(seconds(1));
+    }
+
+    // Wait for some time.
+    sleep_for(seconds(5));
+
+    {
+        // Mission complete. Command RTL to go home.
+        std::cout << "Commanding RTL..." << std::endl;
+        const Action::Result result = action->return_to_launch();
+        if (result != Action::Result::SUCCESS) {
+            std::cout << "Failed to command RTL (" << Action::result_str(result) << ")" << std::endl;
+        } else {
+            std::cout << "Commanded RTL." << std::endl;
+        }
+    }
+
+    return 0;
+}
+
+inline void handle_action_err_exit(Action::Result result, const std::string &message)
+{
+    if (result != Action::Result::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message << Action::result_str(
+                      result) << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+inline void handle_mission_err_exit(Mission::Result result, const std::string &message)
+{
+    if (result != Mission::Result::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message << Mission::result_str(
+                      result) << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+
+// Handles connection result
+inline void handle_connection_err_exit(DroneCore::ConnectionResult result,
+                                       const std::string &message)
+{
+    if (result != DroneCore::ConnectionResult::SUCCESS) {
+        std::cerr << ERROR_CONSOLE_TEXT << message
+                  << DroneCore::connection_result_str(result)
+                  << NORMAL_CONSOLE_TEXT << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+```
+
+[qgroundcontrol_sample.plan](https://github.com/dronecore/DroneCore/blob/{{ book.github_branch }}/example/fly_qgc_mission/qgroundcontrol_sample.plan)
+
+```
+
```

Comment:
This is empty while I wait to see where the example plan file lives - https://github.com/dronecore/DroneCore/issues/272

If it stays in the plugin then I update the cpp file and delete this block.